lib/talkable/middleware.rb

```diff
@@ -8,10 +8,13 @@ def initialize(app)
     end
 
     def call(env)
-      uuid = talkable_visitor_uuid(env)
+      request = Rack::Request.new(env)
+      uuid = talkable_visitor_uuid(request)
 
       result = Talkable.with_uuid(uuid) do
-        @app.call(env)
+        Talkable.with_request_url(request.url) do
```

Comment:
Let's add only one entry to the caller trace. Most cool gems optimize by that parameter because it will show up in every single exception raised by the app.